addons/hr_holidays/models/hr_employee.py

```diff
@@ -513,11 +513,13 @@ def _get_consumed_leaves(self, leave_types, target_date=False, ignore_future=Fal
                                 datetime.combine(allocation.date_to, time.max)
                                 if allocation.date_to else leave.date_to
                             ).replace(tzinfo=pytz.UTC)
-                            duration_info = employee._get_calendar_attendances(interval_start, interval_end)
-                            max_allowed_duration = min(
-                                duration_info['hours' if leave_unit == 'hours' else 'days'],
-                                leave_type_data[allocation]['virtual_remaining_leaves']
-                            )
+                            max_allowed_duration = leave_type_data[allocation]['virtual_remaining_leaves']
+                            if employee.resource_calendar_id:
+                                duration_info = employee._get_calendar_attendances(interval_start, interval_end)
```

Comment:
maybe we could modify `_get_calendar_attendances` instead (and return eg `self.resource_calendar_id._get_attendance_intervals_days_data({})`  if there is no resource calendar)

but in effect this should result in the same thing since this seems to be the only call of this method

Reply:
This doesn't seem to work because `self.resource_calendar_id._get_attendance_intervals_days_data({})` returns `{'days': 0, 'hours': 0}`, which means the result of the `min` call will always be 0, leading to a `"There is no valid allocation to cover that request."` validation error.